### ISSUE TITLE
nixosTests.ceph: ceph-multi-node.nix test fails

### ISSUE THREAD

**Issue by B4dM4n** [NixOS/nixpkgs#77378]
## Issue description

The `ceph-multi-node.nix` test currently fails at [L146](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/ceph-multi-node.nix#L146) with:
```log
osd0: must succeed: cp /tmp/shared/ceph.client.admin.keyring /etc/ceph
osd0 # cp: cannot stat '/tmp/shared/ceph.client.admin.keyring': No such file or directory
osd0: output:
error: command `cp /tmp/shared/ceph.client.admin.keyring /etc/ceph` failed (exit code 1)
```

Bisecting showed that ef5bc381f969205493897b121ce36ecd7bdc1974 is the first failing commit. It changed the `/tmp/shared/` host path and therefore made `/tmp/shared/` not being shared between machines.

Is this intentional? How should files be shared between machines when running under the python driver?

### Steps to reproduce

```
nix build --no-link -f '<nixpkgs/nixos/tests/ceph-multi-node.nix>'
```

<details>
  <summary>last log lines</summary>
  
  ```log
monA: must succeed: sudo -u ceph ceph-authtool --create-keyring /etc/ceph/ceph.client.admin.keyring --gen-key -n client.admin --cap mon 'allow *' --cap osd 'allow *' --cap mds 'allow *' --cap mgr 'allow *'
monA # [   11.097872] sudo[643]:     root : TTY=hvc0 ; PWD=/tmp ; USER=ceph ; COMMAND=/run/current-system/sw/bin/ceph-authtool --create-keyring /etc/ceph/ceph.client.admin.keyring --gen-key -n client.admin --cap mon allow * --cap osd allow * --cap mds allow * --cap mgr allow *
monA # [   11.101214] sudo[643]: pam_unix(sudo:session): session opened for user ceph by (uid=0)
monA # [   16.142940] sudo[643]: pam_unix(sudo:session): session closed for user ceph
(5.06 seconds)
monA: must succeed: sudo -u ceph ceph-authtool /tmp/ceph.mon.keyring --import-keyring /etc/ceph/ceph.client.admin.keyring
monA # [   16.156449] sudo[647]:     root : TTY=hvc0 ; PWD=/tmp ; USER=ceph ; COMMAND=/run/current-system/sw/bin/ceph-authtool /tmp/ceph.mon.keyring --import-keyring /etc/ceph/ceph.client.admin.keyring
monA # [   16.159666] sudo[647]: pam_unix(sudo:session): session opened for user ceph by (uid=0)
monA # [   16.192641] sudo[647]: pam_unix(sudo:session): session closed for user ceph
(0.05 seconds)
monA: must succeed: monmaptool --create --add a 192.168.1.1 --fsid 066ae264-2a5d-4729-8001-6ad265f50b03 /tmp/monmap
(0.04 seconds)
monA: must succeed: sudo -u ceph ceph-mon --mkfs -i a --monmap /tmp/monmap --keyring /tmp/ceph.mon.keyring
monA # [   16.241369] sudo[654]:     root : TTY=hvc0 ; PWD=/tmp ; USER=ceph ; COMMAND=/run/current-system/sw/bin/ceph-mon --mkfs -i a --monmap /tmp/monmap --keyring /tmp/ceph.mon.keyring
monA # [   16.244113] sudo[654]: pam_unix(sudo:session): session opened for user ceph by (uid=0)
monA # [   16.516939] sudo[654]: pam_unix(sudo:session): session closed for user ceph
(0.29 seconds)
monA: must succeed: sudo -u ceph mkdir -p /var/lib/ceph/mgr/ceph-a/
monA # [   16.529684] sudo[679]:     root : TTY=hvc0 ; PWD=/tmp ; USER=ceph ; COMMAND=/run/current-system/sw/bin/mkdir -p /var/lib/ceph/mgr/ceph-a/
monA # [   16.533787] sudo[679]: pam_unix(sudo:session): session opened for user ceph by (uid=0)
monA # [   16.542540] sudo[679]: pam_unix(sudo:session): session closed for user ceph
(0.03 seconds)
monA: must succeed: sudo -u ceph touch /var/lib/ceph/mon/ceph-a/done
monA # [   16.554479] sudo[681]:     root : TTY=hvc0 ; PWD=/tmp ; USER=ceph ; COMMAND=/run/current-system/sw/bin/touch /var/lib/ceph/mon/ceph-a/done
monA # [   16.556898] sudo[681]: pam_unix(sudo:session): session opened for user ceph by (uid=0)
monA # [   16.562355] sudo[681]: pam_unix(sudo:session): session closed for user ceph
(0.02 seconds)
monA: must succeed: systemctl start ceph-mon-a
monA # [   16.578202] systemd[1]: Started Ceph MON daemon a.
(0.02 seconds)
monA: must succeed: ceph mon enable-msgr2
monA # [   16.708693] ceph-mon[685]: 2020-01-09 13:30:52.592 7f3f4c571380 -1 mon.a@0(electing) e0 failed to get devid for : fallback method has serial ''but no model
monA # [   16.746487] ceph-mon[685]: 2020-01-09 13:30:52.630 7f3f3d7f2700 -1 mon.a@0(electing) e1 failed to get devid for : fallback method has serial ''but no model
(0.58 seconds)
monA: must succeed: ceph -s | grep 'mon: 1 daemons'
monA # [   17.313425] ceph-mon[685]: 2020-01-09 13:30:53.197 7ffbcf308380 -1 mon.a@0(electing) e2 failed to get devid for : fallback method has serial ''but no model
(0.30 seconds)
monA: must succeed: ceph auth get-or-create mgr.a mon 'allow profile mgr' osd 'allow *' mds 'allow *' > /var/lib/ceph/mgr/ceph-a/keyring
(0.29 seconds)
monA: must succeed: systemctl start ceph-mgr-a
monA # [   17.841821] systemd[1]: Started Ceph MGR daemon a.
(0.02 seconds)
monA: waiting for success: ceph -s | grep 'quorum a'
(0.40 seconds)
monA: waiting for success: ceph -s | grep 'mgr: a(active,'
monA # [   21.229151] nscd[514]: 514 monitored file `/etc/resolv.conf` changed (mtime)
monA # [   21.230131] nscd[514]: 514 monitoring file `/etc/resolv.conf` (6)
monA # [   21.231326] nscd[514]: 514 monitoring directory `/etc` (2)
osd1 # [   21.208876] nscd[508]: 508 monitored file `/etc/resolv.conf` changed (mtime)
osd1 # [   21.210336] nscd[508]: 508 monitoring file `/etc/resolv.conf` (6)
osd1 # [   21.211461] nscd[508]: 508 monitoring directory `/etc` (2)
osd0 # [   21.342761] nscd[507]: 507 monitored file `/etc/resolv.conf` changed (mtime)
osd0 # [   21.344656] nscd[507]: 507 monitoring file `/etc/resolv.conf` (6)
osd0 # [   21.345737] nscd[507]: 507 monitoring directory `/etc` (2)
monA # [   23.229359] nscd[514]: 514 checking for monitored file `/etc/netgroup': No such file or directory
osd1 # [   23.208529] nscd[508]: 508 checking for monitored file `/etc/netgroup': No such file or directory
osd0 # [   23.342680] nscd[507]: 507 checking for monitored file `/etc/netgroup': No such file or directory
(7.42 seconds)
monA: must succeed: cp /etc/ceph/ceph.client.admin.keyring /tmp/shared
(0.01 seconds)
osd0: must succeed: cp /tmp/shared/ceph.client.admin.keyring /etc/ceph
osd0 # cp: cannot stat '/tmp/shared/ceph.client.admin.keyring': No such file or directory
osd0: output:
error: command `cp /tmp/shared/ceph.client.admin.keyring /etc/ceph` failed (exit code 1)
cleaning up
killing monA (pid 9)
killing osd0 (pid 25)
killing osd1 (pid 41)
(0.00 seconds)
builder for '/nix/store/v214cjhwzwyajamhyy1i75qk73yxacrk-vm-test-run-basic-multi-node-ceph-cluster.drv' failed with exit code 1
error: build of '/nix/store/v214cjhwzwyajamhyy1i75qk73yxacrk-vm-test-run-basic-multi-node-ceph-cluster.drv' failed
 ```
</details>

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.1, NixOS, 20.03pre206632.b0bbacb5213 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(root): `"nixos-20.03pre206632.b0bbacb5213"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

CC @7c6f434c @flokli

**Comment by flokli** [comment]
See https://github.com/NixOS/nixpkgs/pull/74077#issuecomment-571459974 cc @srhb

**Comment by B4dM4n** [comment]
Fixed by #77587
